### PR TITLE
OpenSSL dependency update to support v3.x

### LINKED ISSUE #339: OpenSSL dependency update to support v3.x
### Short description of issue

Need to update this to support clients building on machines without OpenSSL 1.x libs.

### Package version (and git commit)

`head`

### Operating System / Version

All

### PR DESCRIPTION
**Pull request by yaleman**
Fixes #339

- [x] cargo test has been run and passes
- [ ] documentation has been updated with relevant examples (if relevant)

### CODE REVIEW

**Review by Firstyear — DISMISSED** [review]
No issues from me, I'll let @micolous check too.

**Comment by Firstyear** [comment]
```
libopenssl-devel-3.1.2-1.1.noarch
```
Seems to be working on my tumbleweed machine? Also confirmed it works on my mac with 3.1.2 as well. 

Do you potentially have a second openldap/libressl present?

**Comment by dmulder** [comment]
> Do you potentially have a second openldap/libressl present?

Yes, I have libressl.

**Comment by Firstyear** [comment]
Ahh that'll be it. libressl is broken wrt to openssl-rs normally, so it's likely a missing feature/flag/const in libressl.

File: Makefile
Comment (micolous) [inline Makefile:1]: Please put this in a separate PR, it seems unrelated to this proposal.

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+---
```
Comment (micolous) [inline .github/dependabot.yml:1]: Please put this in a separate PR after this one; we should get OpenSSL update out of the way first.

File: Cargo.toml [hunk 2]
```diff
@@ -34,9 +34,7 @@ members = [
 ]
 
 # Due to --cfg=web_sys_unstable_apis
-exclude = [
```
Comment (micolous) [inline Cargo.toml:37]: Please split formatting changes into a separate PR before this one.

It makes it difficult to review your changes.

File: fido-hid-rs/src/lib.rs [hunk 3]
```diff
@@ -52,7 +52,7 @@ const FIDO_USAGE_PAGE: u16 = 0xf1d0;
 const FIDO_USAGE_U2FHID: u16 = 0x01;
 const HID_RPT_SIZE: usize = 64;
 const HID_RPT_SEND_SIZE: usize = HID_RPT_SIZE + 1;
-const U2FHID_TRANS_TIMEOUT: i32 = 3000;
```
Comment (micolous) [inline fido-hid-rs/src/lib.rs:55]: This change is not related to the PR, please split this.

**Review by micolous — CHANGES_REQUESTED** [review]

**Comment by yaleman** [comment]
@micolous I've reverted what fmt changes I can and I don't have enough git-foo to convince the others to calm themselves, I hope that's OK.

**Review by micolous — APPROVED** [review]

**Comment by micolous** [comment]
Thanks! CI failures on MSRV look like a separate issue not triggered by this change, I've opened https://github.com/kanidm/webauthn-rs/issues/346 about this.

**Pull request merged**